lib/config/migration.ts

```diff
@@ -237,21 +237,28 @@ export function migrateConfig(
         } else {
           migratedConfig.semanticCommitScope = null;
         }
-      } else if (key === 'extends' && is.array<string>(val)) {
-        for (let i = 0; i < val.length; i += 1) {
-          if (val[i] === 'config:application' || val[i] === ':js-app') {
+      } else if (
+        key === 'extends' &&
+        (is.array<string>(val) || is.string(val))
+      ) {
+        if (is.string(migratedConfig.extends)) {
+          migratedConfig.extends = [migratedConfig.extends];
+          isMigrated = true;
+        }
+        const presets = migratedConfig.extends;
+        for (let i = 0; i < presets.length; i += 1) {
+          let preset = presets[i];
+          if (preset === 'config:application' || preset === ':js-app') {
             isMigrated = true;
-            migratedConfig.extends[i] = 'config:js-app';
-          } else if (val[i] === ':library' || val[i] === 'config:library') {
+            preset = 'config:js-app';
+          } else if (preset === ':library' || preset === 'config:library') {
             isMigrated = true;
-            migratedConfig.extends[i] = 'config:js-lib';
-          } else if (is.string(val[i]) && val[i].startsWith(':masterIssue')) {
+            preset = 'config:js-lib';
+          } else if (is.string(preset) && preset.startsWith(':masterIssue')) {
```

Comment:
shouldn't the `is.string` check come first, that should be an error if it is not a string.

Reply:
Seems like we don't validate preset values in general

Reply:
I think they're validated later when we attempt to resolve them